components/table/index.jsx

```diff
@@ -173,7 +173,7 @@ class InfiniteTable extends React.PureComponent {
 
     this.setState(newState)
 
-    const data = await pages.slice(0, size)
+    const data = await pages.slice(0, newSize)
```

Comment:
Where do you get `newSize`?

Reply:
How is it not clear? For example if the value of newSize is 1000 then `pages` instance will check how many records. If less<1000 it loads next page and checks if >=1000. If not it load another batch and checks again.

So, in this case, the newSize is reset to 100 records, i.e. only one page is loaded.